src/impression.js

```diff
@@ -102,14 +104,21 @@ function trackImpressionFromClickParam(win, viewer, resolveImpression) {
   }
 
   viewer.whenFirstVisible().then(() => {
+    viewer.sendMessageAwaitResponse('getReplaceUrl').then(response => {
+      if (response) {
+        replaceUrl(win, response);
+      }
+      resolveImpression();
+    }, () => {});
```

Comment:
It's possible (though unlikely) for the replaceUrl to be invalid. This will happen if the replaceUrl provided doesn't match the original origin of the AMP document -- or, if there's a redirect, the final origin of the AMP document (matching either is acceptable).

It's probably safe to ignore but figured I'd flag to be safe.

Reply:
Since viewer won't send ampdoc the error information. `#sendMessageAwaitResponse` only throws messaging channel errors. Not related to replaceUrl, so I think we would prefer ignore.